### PR TITLE
Pathlib /unit/mainwindow

### PR DESCRIPTION
**Pull request by Lembrun**
#176

### CODE REVIEW

File: tests/unit/mainwindow/test_prompt.py [hunk 1]
```diff
@@ -53,26 +53,26 @@ def _get_prompt_func(path):
...
     ])
-    def test_simple_completion(self, tmpdir, get_prompt, steps, where,
+    def test_simple_completion(self, tmp_path, get_prompt, steps, where,
                                subfolder):
         """Simply trying to tab through items."""
-        testdir = tmpdir / 'test'
+        testdir = tmp_path / 'test'
         for directory in 'abc':
-            (testdir / directory).ensure(dir=True)
+            (testdir / directory).mkdir(exist_ok=True, parents=True)
```
Comment (The-Compiler) [inline tests/unit/mainwindow/test_prompt.py:61]: Is the `exist_ok=True` needed here? We should get an empty `tmp_path` for every time this test runs, so I don't think it should ever exist?

File: tests/unit/mainwindow/test_prompt.py [hunk 2]
```diff
@@ -53,26 +53,26 @@ def _get_prompt_func(path):
...
 
-    def test_backspacing_path(self, qtbot, tmpdir, get_prompt):
+    def test_backspacing_path(self, qtbot, tmp_path, get_prompt):
         """When we start deleting a path we want to see the subdir."""
-        testdir = tmpdir / 'test'
+        testdir = tmp_path / 'test'
 
         for directory in ['bar', 'foo']:
-            (testdir / directory).ensure(dir=True)
+            (testdir / directory).mkdir(exist_ok=True, parents=True)
```
Comment (The-Compiler) [inline tests/unit/mainwindow/test_prompt.py:75]: Is the `exist_ok=True` needed here?

**Review by The-Compiler — CHANGES_REQUESTED** [review]

File: tests/unit/mainwindow/test_prompt.py (hunk 2 above)
Comment (Lembrun) [reply tests/unit/mainwindow/test_prompt.py:75]: You're indeed right

**Comment by The-Compiler** [comment]
Thanks!

**Pull request merged**